fabtests/common/shared.c

```diff
@@ -2922,7 +2929,7 @@ void ft_csusage(char *name, char *desc)
 	FT_PRINT_OPTS_USAGE("-I <number>", "number of iterations");
 	FT_PRINT_OPTS_USAGE("-Q", "bind EQ to domain (vs. endpoint)");
 	FT_PRINT_OPTS_USAGE("-w <number>", "number of warmup iterations");
-	FT_PRINT_OPTS_USAGE("-S <size>", "specific transfer size or 'all'");
+	FT_PRINT_OPTS_USAGE("-S <size>", "specific transfer size or a range of sizes (syntax r:start,inc,end) or 'all'");
```

Comment:
line length should be limited to 80 characters.  Comment applies to other places in the code.

This would be simpler as a couple new options, rather than creating a complex parsing string.  -E end_size -T size stepping.  If -E is omitted, end_size = start size (-S option).  If -T omitted, stepping is by 1

Reply:
I have updated the changed lines in this PR to be less than 80 characters-- however, I haven't updated rest of the lines in the code that were violating the 80 character constraint (I saw a bunch while using the marker).

Reply:
Regarding using -E and -T for end size and stepping, it would be simpler indeed but there are two caveats (1) The code currently sets FT_OPT_SIZE flag when -S is used with a transfer size. The benchmarks make specific checks if FT_OPT_SIZE is not enabled then a range or all sizes are used. This approach would require modifications to the flag values and the benchmarks if we want to use -S as a range. (2) We are already using -E for out of band address exchange and we didn't want to run out of characters to be used.

Reply:
I picked random characters for the options.  We would obviously need to pick ones that work.  Other issues can easily be solved.  If an end size is set, just clear FT_OPT_SIZE flag, or add a new flag to indicate a specific range is requested instead, or use a different option to specify the starting value.

Reply:
There are not many letters left though.

I did some grep and noticed that fabtests have already used 34,  which means there are only 18 left. Maybe we want to save some for the future?


```
	FT_PRINT_OPTS_USAGE("-B <src_port>", "non default source port number");
	FT_PRINT_OPTS_USAGE("-P <dst_port>", "non default destination port number");
	FT_PRINT_OPTS_USAGE("-s <address>", "source address");
	FT_PRINT_OPTS_USAGE("-b[=<oob_port>]", "enable out-of-band address exchange and "
	FT_PRINT_OPTS_USAGE("-E[=<oob_port>]", "enable out-of-band address exchange only "
	FT_PRINT_OPTS_USAGE("-C <number>", "number of connections to accept before "
	FT_PRINT_OPTS_USAGE("-F <addr_format>", "Address format (default:FI_FORMAT_UNSPEC)");
	FT_PRINT_OPTS_USAGE("-f <fabric>", "fabric name");
	FT_PRINT_OPTS_USAGE("-d <domain>", "domain name");
	FT_PRINT_OPTS_USAGE("-p <provider>", "specific provider name eg sockets, verbs");
	FT_PRINT_OPTS_USAGE("-e <ep_type>", "Endpoint type: msg|rdm|dgram (default:rdm)");
	FT_PRINT_OPTS_USAGE("-U", "run fabtests with FI_DELIVERY_COMPLETE set");
	FT_PRINT_OPTS_USAGE("-M <mode>", "Disable mode bit from test");
	FT_PRINT_OPTS_USAGE("-K", "fork a child process after initializing endpoint");
	FT_PRINT_OPTS_USAGE("-a <address vector name>", "name of address vector");
	FT_PRINT_OPTS_USAGE("-h", "display this help output");
	FT_PRINT_OPTS_USAGE("-f <fabric>", "fabric name");
	FT_PRINT_OPTS_USAGE("-d <domain>", "domain name");
	FT_PRINT_OPTS_USAGE("-p <provider>", "specific provider name eg sockets, verbs");
	FT_PRINT_OPTS_USAGE("-D <device_iface>", "Specify device interface: "
	FT_PRINT_OPTS_USAGE("-i <device_id>", "Specify which device to use (default: 0)");
	FT_PRINT_OPTS_USAGE("-H", "Enable provider FI_HMEM support");
	FT_PRINT_OPTS_USAGE("-h", "display this help output");
	FT_PRINT_OPTS_USAGE("-I <number>", "number of iterations");
	FT_PRINT_OPTS_USAGE("-Q", "bind EQ to domain (vs. endpoint)");
	FT_PRINT_OPTS_USAGE("-w <number>", "number of warmup iterations");
	FT_PRINT_OPTS_USAGE("-S <size>", "specific transfer size or 'all'");
	FT_PRINT_OPTS_USAGE("-l", "align transmit and receive buffers to page size");
	FT_PRINT_OPTS_USAGE("-m", "machine readable output");
	FT_PRINT_OPTS_USAGE("-D <device_iface>", "Specify device interface: "
	FT_PRINT_OPTS_USAGE("-i <device_index>", "Index of the device to use");
	FT_PRINT_OPTS_USAGE("-t <type>", "completion type [queue, counter]");
	FT_PRINT_OPTS_USAGE("-c <method>", "completion method [spin, sread, fd, yield]");
	FT_PRINT_OPTS_USAGE("-h", "display this help output");
```

Reply:
I think we should create a new argument called 

```
--msg_size=(500,700,4]
```

and implement custom parser logic (or use an existing library) to translate the provided CLI range string into python range object 

```
msg_size = range(504,704,4)
```

This way the proposed `--msg_size` argument will be extensible to all fabtests, and offer more clarity on its purpose.

Reply:
We do want to have these tests written in C for performance reasons. Having more flexibility to parse fabtest arguments by using string instead of int might be useful in the long run.

For now I have used sscanf to parse the range using the -S option. We thought about introducing a different '-R' option to parse this range but it would be counter-intuitive to provide together "-S all -R [start, inc, end]" for message sizes.